projects/rocfft/library/src/plan.cpp

```diff
@@ -2103,6 +2121,225 @@ void rocfft_plan_t::GlobalTransposeP2P(size_t                     elem_size,
     }
 }
 
+// global transpose using MPI sub-communicators
+void rocfft_plan_t::GlobalTransposeA2ASubcomm(size_t                     elem_size,
+                                              const rocfft_field_t&      inField,
+                                              const rocfft_field_t&      outField,
+                                              std::vector<BufferPtr>&    input,
+                                              std::vector<BufferPtr>&    output,
+                                              const std::vector<size_t>& inputAntecedents,
+                                              std::vector<size_t>&       outputItems,
+                                              const std::string&         itemGroup,
+                                              MPI_Comm_wrapper_t&&       subcomm)
+{
+#ifdef ROCFFT_MPI_ENABLE
+    int subcomm_rank = -1, subcomm_size = -1;
+    MPI_Comm_rank(subcomm, &subcomm_rank);
+    MPI_Comm_size(subcomm, &subcomm_size);
+
+    // map subcomm-local rank <-> global rank for this pencil
+    // for all ranks, set up a vector:
+    //   global_rank_of_subcomm_rank[local_subcomm_rank] = global rank
+    //   subcomm_rank_of_global_rank[global_rank] = subcomm-local rank, or -1 if not present
+    std::vector<int> global_rank_of_subcomm_rank(subcomm_size, -1);
+    std::vector<int> subcomm_rank_of_global_rank(get_local_comm_size(), -1);
+
+    // build mapping by MPI_Allgather (gathering global rank from all subcomm ranks)
+    int my_global_rank = get_local_comm_rank();
+    MPI_Allgather(
+        &my_global_rank, 1, MPI_INT, global_rank_of_subcomm_rank.data(), 1, MPI_INT, subcomm);
+    for(int i = 0; i < subcomm_size; ++i)
+        subcomm_rank_of_global_rank[global_rank_of_subcomm_rank[i]] = i;
+
+    // only communicate with the ranks in this subcomm
+    std::optional<int> local_send_device;
+    std::optional<int> local_recv_device;
+
+    size_t              cumulative_send_elems = 0;
+    size_t              cumulative_recv_elems = 0;
+    std::vector<size_t> send_offsets(subcomm_size, 0);
+    std::vector<size_t> send_counts(subcomm_size, 0);
+    std::vector<size_t> recv_offsets(subcomm_size, 0);
+    std::vector<size_t> recv_counts(subcomm_size, 0);
+
+    std::vector<size_t> pack_ops;
+    std::vector<size_t> unpack_ops;
+
+    // packing/unpacking only within subcomm
+    for(size_t inBrickIdx = 0; inBrickIdx < inField.bricks.size(); ++inBrickIdx)
+    {
+        const auto& inBrick       = inField.bricks[inBrickIdx];
+        int         inRank        = inBrick.location.comm_rank;
+        int         inSubcommRank = subcomm_rank_of_global_rank[inRank];
```

Comment:
```suggestion
        const int         inRank        = inBrick.location.comm_rank;
        const int         inSubcommRank = subcomm_rank_of_global_rank[inRank];
```
and other similar cases below.

Reply:
This is fixed in the new commit.